### PR TITLE
935 - Polls front

### LINKED ISSUE #935: Participation polls front
Tasks:

- [x] Backend code
- [x] Tests
- [x] AJAX call to submit poll - Callbacks for success/error
- [x] When clicking on (X), redirecto to polls#index
- [x] Show share in social networks slide after successfully submitting poll
- [x] Create new layout without header and footer
- [x] Check user is logged in before letting him answer a poll
- [x] If user is not logged in, show login modal
- [x] Link poll fron admin ("View poll" link)
- [x] Disable next question button until user answers question
- [x] Validations at form object level to check all questions were answered
- [x] Let answers to open questions be blank
- [x] In polls index, deactivate link if already participated in poll

Pending:

- [ ] CSS for disabled next question button
- [x] View poll links in admin/process/process_id/polls table

For the future:

- [ ] Save poll state for when an error occurs
- [ ] Check user has right permissions to answer a poll (census integration pending)
- [ ] Button for going back between the questions

Pending edge cases:

- [x] Polls with 1 question are buggy (refactor JS also)
- [x] Check links from admin (preview tokens etc.)

### PR DESCRIPTION
**Pull request by amiedes**
Connects to #935, closes #935 and #833.

### What does this PR do?

Implements UI for participating in polls

### Pending tasks:

Little CSS adjustments are needed in the following things:

- [x] Fix CSS of close poll button (x).
- [x] Create a "disabled" style for the next question button.

And a few nice-to-haves for the future (issue #959):

- [ ] Save poll state for when an error occurs
- [ ] Check user has right permissions to answer a poll (census integration pending)
- [ ] Button for going back between the questions

### Does this PR changes any configuration file?

No

### CODE REVIEW

File: app/controllers/gobierto_participation/processes/poll_answers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          render :new, layout: 'gobierto_participation/layouts/application_simple'
+        else
+          redirect_to(
+            gobierto_participation_process_polls_path(current_process.slug),
+            alert: t('.already_answered')
+          )
+        end
+      end
+
+      def create
```
Comment (ferblape) [inline app/controllers/gobierto_participation/processes/poll_answers_controller.rb]: Why not a before_filter? I.e. (from BudgetConsultations)

```ruby
      before_action :authenticate_user!
      before_action { verify_user_in!(current_site) }
```

File: app/controllers/gobierto_participation/processes/base_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+module GobiertoParticipation
+  module Processes
+    class BaseController < GobiertoParticipation::ApplicationController
+
+      helper_method :current_process
+
+      protected
+
+      def current_process
+        @current_process ||= current_site.processes.find_by_slug!(params[:process_id])
```
Comment (ferblape) [inline app/controllers/gobierto_participation/processes/base_controller.rb]: You are not checking if the process is active.

File: app/views/layouts/_head.html.erb [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<head>
```
Comment (ferblape) [inline app/views/layouts/_head.html.erb:1]: 👍

**Review by ferblape — CHANGES_REQUESTED** [review]
And you forgot this translation:

screen shot 2017-09-20 at 08 47 05

Once you have fixed all the issues described please:

1. Assign the PR to Hugo, to fix the couple of things you have mentioned
2. Create a new issue for the improvements

👍

**Comment by amiedes** [comment]
Hey @Crashillo, can you checkout this PR and fix the CSS issues?

We need to fix the upper right close button:

image

And to add some styling to the "Next question" button when it has the `disabled` attribute set:

image

**Comment by amiedes** [comment]
Also I did a couple of hacks/workarounds. I'm not sure if they can stay like this:

image

**Comment by furilo** [comment]
> I'm not sure if they can stay like this

No, they can't. 

@Crashillo let's wait until I give you the Gobierto intro.

**Comment by ferblape** [comment]
@amiedes I have found something you need to fix, in this process in staging: http://participacion.gobify.net/processes/xadcasd

1. It doesn't have the polls stage activated, so it shouldn't show the link to "Encuestas" in the frontend
2. That action should have a kind of filter, so if you visit the url it should redirect to the home of the process
3. That URL doesn't show in the breadcrumb the name of the process

screen shot 2017-09-21 at 08 50 08

**Review by ferblape — APPROVED** [review]

**Comment by ferblape** [comment]
@amiedes could you take care of solving the conflicts and merging this branch to master?

**Comment by amiedes** [comment]
Sure!

**Pull request merged**